tlslite/tlsrecordlayer.py

```diff
@@ -722,6 +736,40 @@ def _getMsg(self, expectedType, secondaryType=None, constructorType=None):
                                 yield result
                             continue
 
+                    # If we received a heartbeat request and heartbeat
+                    # extension was negotiated
+                    if recordHeader.type == ContentType.heartbeat and \
+                            self.heartbeatSupported:
+                        try:
+                            heartbeat_message = Heartbeat().parse(p)
+                            # If we received heartbeat request, then we
+                            # response with response create from request
+                            if heartbeat_message.message_type == \
+                                    HeartbeatMessageType.heartbeat_request:
+                                if not self.heartbeatCanReceive:
+                                    for result in self._sendError(
+                                            AlertDescription.
+                                            unexpected_message,
+                                            "Received heartbeat_request to " +
+                                            "peer_not_allowed_to_send mode"):
+                                        yield result
+                                else:
+                                    heartbeat_response = heartbeat_message.\
+                                        createResponse()
+                                for result in self._sendMsg(
+                                        heartbeat_response):
+                                    yield result
+                            # If we received heartbeat response, then we
+                            # check, if its payload is same as payload of
+                            # request we sent
+                            elif heartbeat_message.message_type == \
+                                    HeartbeatMessageType.heartbeat_response:
+                                heartbeat_message.checkResponse(
```

Comment:
either just payload, or the whole `hearbeat_message`

the parameter of that callback would then be just that - the response received.

it can look like in any possible way after that, quick example:
```
sent_heartbeats = []
heartbeat_count = 0


def heartbeat_cb(message):
   no, payload = next((no, payload) for no, payload in sent_requests if payload == message.payload), (None, None))
   if no is None:
     raise ValueError("Received unexpected heartbeat response")
  sent_requests.remove((no, payload))
  print("Got response to hearbeat # {0}".format(no))
  
def main():
   ...
   handshake_settings = HandshakeSettings()
   handshake_settings.heartbeat_callback = heartbeat_cb
   socket.handshake(handshake_settings)

   payload = getRandomBytes(32)
   heartbeat_count += 1
   sent_heartbeats.append((heartbeat_count, payload))
   socket.sentHeartbeatRequest(payload)
   ...
```

(sorry for typos, mismatched object names and method names, writing from memory, hope it's useful)